### ISSUE TITLE
train magicpoint：Loss is difficult to decline

### ISSUE THREAD

**Issue by xiaobo8310** [rpautrat/SuperPoint#294]
Hello

I reproduce your magicpoint code and output the semi (b,65,h/8,w/8) according to the official network structure.

Then find the loss (the loss calculation in your code does not seem exactly the same as in the paper) and optimize it with adam.

But the loss has not gone down

[code.txt](https://github.com/rpautrat/SuperPoint/files/11355582/code.txt)

**Comment by rpautrat** [comment]
Hi,

We did slightly modify the original descriptor loss, as this formulation gave us better results.

I cannot help you much with your code, as it is a reimplementation and is out of the scope of this repo, but you might want to check this Pytorch implementation of SuperPoint: https://github.com/eric-yyjau/pytorch-superpoint. Maybe it can help.
